### PR TITLE
[V5] Allow revokePermissionTo to accept Permission[]

### LINKED ISSUE #2010: revokePermissionTo incorrect behaviour with Contracts\Permission[] argument
**Description**
When called with an argument of Permission[] rather than string[] or int[] when revoking multiple permissions, revokePermissionsTo does not revoke any permissions.

givePermissionTo does work correctly with an argument of Permission[] though, so this is a bit confusing.

**Versions**
- spatie/laravel-permission package version: 5.5.0
- laravel/framework package version: 8.79.0

PHP version: 8.1

**To Reproduce**
1. Assign permissions to model with HasPermissions trait
`$u->givePermissionTo([Permission::firstWhere('name', 'view articles'), Permission::firstWhere('name', 'edit articles'), Permission::firstWhere('name', 'delete articles')])`
2. Revoke permissions
`$u->revokePermissionTo([Permission::firstWhere('name', 'edit articles'), Permission::firstWhere('name', 'delete articles')])`
3. Check model's permissions. No permissions will have been revoked.

**Expected Behavior**
The specified permissions are removed from the model.

**Environment:**
 - OS: Linux x86_64
 - Version 5.10.76

### PR DESCRIPTION
**Pull request by erikn69**
Another alternative for #2011 (With tests and without breaking changes)
Closes #2010

### CODE REVIEW

File: src/Traits/HasPermissions.php [hunk 1]
```diff
@@ -426,6 +426,10 @@ protected function getStoredPermission($permissions)
         }
 
         if (is_array($permissions)) {
+            $permissions = array_map(function ($permision) use ($permissionClass) {
```
Comment (Restingo) [inline src/Traits/HasPermissions.php:429]:
```suggestion
            $permissions = array_map(function ($permission) use ($permissionClass) {
```

File: src/Traits/HasPermissions.php [hunk 2]
```diff
@@ -426,6 +426,10 @@ protected function getStoredPermission($permissions)
...
+                return is_a($permision, get_class($permissionClass)) ? $permision->name : $permision;
```
Comment (Restingo) [inline src/Traits/HasPermissions.php:430]:
```suggestion
                return is_a($permission, get_class($permissionClass)) ? $permission->name : $permission;
```

**Review by Restingo — COMMENTED** [review]
Typo

File: src/Traits/HasPermissions.php (hunk 1 above)
Comment (erikn69) [reply src/Traits/HasPermissions.php]: Thanks

**Comment by angeljqv** [comment]
@freekmurze ping
Closes https://github.com/spatie/laravel-permission/issues/2010

**Comment by freekmurze** [comment]
Thanks!

**Pull request merged**